pkg/primitive/concat.go

```diff
@@ -0,0 +1,64 @@
+package primitive
+
+import (
+	"fmt"
+	"strings"
+)
+
+// Concat represents the concat primitive
+type Concat struct{}
+
+func (c *Concat) Name() string {
+	return "/gnd/concat"
+}
+
+func (c *Concat) Execute(args []interface{}) (interface{}, error) {
+	if len(args) == 0 {
+		return nil, fmt.Errorf("concat expects at least 1 argument, got 0")
+	}
+
+	// If only one argument, return it directly
+	if len(args) == 1 {
+		return args[0], nil
+	}
+
+	// Check if first argument is an array
+	if _, ok := args[0].([]interface{}); ok {
+		return c.concatArrays(args)
+	}
+
+	// Otherwise treat as string concatenation
+	return c.concatStrings(args)
+}
+
+// isArray checks if a string represents an array
+func isArray(s string) bool {
+	return len(s) >= 2 && s[0] == '[' && s[len(s)-1] == ']'
+}
+
+// concatArrays concatenates arrays and non-array values
+func (c *Concat) concatArrays(args []interface{}) (interface{}, error) {
+	var result []interface{}
+
+	// Process each argument
+	for _, arg := range args {
+		if arr, ok := arg.([]interface{}); ok {
+			result = append(result, arr...)
+		} else if str, ok := arg.(string); ok {
+			result = append(result, str)
+		} else {
+			result = append(result, fmt.Sprintf("%v", arg))
+		}
+	}
+
+	return result, nil
```

Comment:
We could implement `reduce`, `append`, and `isArray` and do this in pure GND syntax.

Reply:
Added task for this https://github.com/hyperifyio/gnd/issues/67